src/WebJobs.Script.Abstractions/Description/Binding/BindingMetadata.cs

```diff
@@ -78,24 +92,29 @@ public bool IsReturn
         /// <returns>The new <see cref="BindingMetadata"/> instance.</returns>
         public static BindingMetadata Create(JObject raw)
         {
-            string bindingDirectionValue = (string)raw["direction"];
-            string connection = (string)raw["connection"];
-            string bindingType = (string)raw["type"];
-            BindingDirection bindingDirection = default(BindingDirection);
-
-            if (!string.IsNullOrEmpty(bindingDirectionValue) &&
-                !Enum.TryParse<BindingDirection>(bindingDirectionValue, true, out bindingDirection))
+            if (raw is null)
             {
-                throw new FormatException(string.Format(CultureInfo.InvariantCulture, "'{0}' is not a valid binding direction.", bindingDirectionValue));
+                throw new ArgumentNullException(nameof(raw));
             }
 
-            BindingMetadata bindingMetadata = raw.ToObject<BindingMetadata>();
-            bindingMetadata.Type = bindingType;
-            bindingMetadata.Direction = bindingDirection;
-            bindingMetadata.Connection = connection;
-            bindingMetadata.Raw = raw;
+            try
+            {
+                BindingMetadata bindingMetadata = raw.ToObject<BindingMetadata>();
+                bindingMetadata.Raw = raw;
+
+                return bindingMetadata;
+            }
+            catch (Exception ex)
+            {
+                string bindingDirectionValue = (string)raw["direction"];
+                if (!string.IsNullOrEmpty(bindingDirectionValue) &&
+                    !Enum.TryParse<BindingDirection>(bindingDirectionValue, true, out BindingDirection bindingDirection))
+                {
+                    throw new FormatException(string.Format(CultureInfo.InvariantCulture, "'{0}' is not a valid binding direction.", bindingDirectionValue));
+                }
 
-            return bindingMetadata;
+                throw ex;
```

Comment:
```suggestion
                throw;
```

`throw ex;` will reset the stack trace on that exception. `throw;` will preserve the stack trace.